### PR TITLE
runInteractiveProcess is deprecated, use createProcess

### PR DESCRIPTION
**Pull request by dopamane**
### Scope:
* According to [process](https://hackage.haskell.org/package/process) library [docs](https://hackage.haskell.org/package/process-1.6.13.2/docs/System-Process.html#g:8), [`runInteractiveProcess`](https://hackage.haskell.org/package/process-1.6.13.2/docs/System-Process.html#v:runInteractiveProcess) is deprecated, use [`createProcess`](https://hackage.haskell.org/package/process-1.6.13.2/docs/System-Process.html#v:createProcess) instead.
* Use [`cleanupProcess`](https://hackage.haskell.org/package/process-1.6.13.2/docs/System-Process.html#v:cleanupProcess) instead of manually closing pipes with `hClose`; see `stop`.
* Minor syntactic cleanup
    * `System.Process` exports `proc`, keep our namespace clear by changing `proc` to `solver`.
    * use `void` in `ackCommandIgnoreErr`
    * use `Prelude.Kore` instead of `Prelude`
    * Cleanup redirection of error pipe to debug logger (this still needs a serious redesign in a later PR, it is not a properly managed process).

PS. I had intended to improve the solver interface by using the bracket-style [`withCreateProcess`](https://hackage.haskell.org/package/process-1.6.13.2/docs/System-Process.html#v:withCreateProcess) to safely open and close the SMT process handle. I've decided to defer that change (commit https://github.com/kframework/kore/pull/2900/commits/ecae94beff39be2eb72dad429e3454fa4b56482f, reversion https://github.com/kframework/kore/pull/2900/commits/2e7e2397c3ed644f6df746203105b92fe00045b2) to a later PR.

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/src/SMT/SimpleSMT.hs [hunk 1]
```diff
@@ -332,22 +335,29 @@ newSolver ::
...
-        X.handle handler $
-            Monad.forever $ do
-                errs <- Text.hGetLine hErr
-                debug (Solver solverHandle logger) errs
+    -- redirect error pipe to debug logger
+    Text.hGetLine hErr >>= debug (Solver solverHandle logger)
+        & Monad.forever
+        & X.handle (\X.SomeException{} -> return ())
+        & forkIO
+        & void
```
Comment (ana-pantilie) [inline kore/src/SMT/SimpleSMT.hs:347]: It took me a while to realize that this just refactors the old code. I think the old code is clearer, I'd probably just factor out the `Monad.forever $ do ...` part and name it `redirectErrorsToLogger`, and also rename the `handler` to `ignoreExceptions` or something of the sort.

The main part of the code will look like:
```suggestion
  void $ forkIO (X.handle ignoreExceptions redirectErrorsToLogger)
```

This way the code is self documenting, removing any need of additional comments.

File: kore/src/SMT/SimpleSMT.hs [hunk 2]
```diff
@@ -210,28 +210,31 @@ import Prelude.Kore hiding (
     not,
     or,
  )
+import qualified Prelude.Kore as Prelude
```
Comment (ana-pantilie) [inline kore/src/SMT/SimpleSMT.hs:213]: Why is this imported qualified?

**Review by ana-pantilie — CHANGES_REQUESTED** [review]

File: kore/src/SMT/SimpleSMT.hs (hunk 2 above)
Comment (dopamane) [reply kore/src/SMT/SimpleSMT.hs:213]: The SimpleSMT module defines some overlapping function names: `not`, `or`, `div`, etc. See `import Prelude.Kore hiding (...` just above line 213 for a complete list. To disambiguate those functions the import needs to be qualified. This change is just replacing `import qualified Prelude as Prelude` with `import qualified Prelude.Kore as Prelude`. Either editorconfig or fourmolu rearranged the imports so the change isn't obvious on Github.

Comment (ana-pantilie) [reply kore/src/SMT/SimpleSMT.hs:213]: Alright, sounds good then!

Comment (dopamane) [reply kore/src/SMT/SimpleSMT.hs:213]: My reasoning for the change from `Prelude` to `Kore.Prelude` is for consistency; we should always prefer to use our custom prelude over the standard prelude.

Comment (ana-pantilie) [reply kore/src/SMT/SimpleSMT.hs:213]: Yeah, definitely.

File: kore/src/SMT/SimpleSMT.hs (hunk 1 above)
Comment (dopamane) [reply kore/src/SMT/SimpleSMT.hs]: Ok, updated in https://github.com/kframework/kore/pull/2900/commits/595b7c2522e525ce9b87870a37c47130c3b299a3. Let me know what you think.

**Review by ana-pantilie — APPROVED** [review]

**Review by andreiburdusa — APPROVED** [review]

**Pull request merged**